### PR TITLE
mrustc: update to 20220904

### PR DESCRIPTION
**Pull request by catap**
#### Description

Update includes fix for 10.14.

It also includes a small patch reorganization that fixed build on 10.7.

And as bonus I've also fixed support of GCC as compiler for mrustc :)

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.7.5 11G63 x86_64
Xcode 4.6.3 4H1503

macOS 10.14.6 18G9323 x86_64
Xcode 11.3.1 11C505

and as `port -s install mrustc mrustc-rust configure.compiler=macports-gcc-12` on

macOS 12.5.1 21G83 x86_64
Xcode 13.4.1 13F100

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by catap** [comment]
Cc: @evanmiller

**Comment by catap** [comment]
Keep it as draft. Maybe upstream had fixed support of arm64.

**Comment by catap** [comment]
Seems that not yet ready

**Comment by pmetzger** [comment]
@catap Is this ready to merge?

**Comment by catap** [comment]
@mascguy may I ask you to review and possible merge this one? :)

**Review by mascguy — APPROVED** [review]
LGTM, I'll merge once the buildbots are idle. (They're presently busy with `gcc-devel`, and will be tied up for at least a few more hours.)

**Comment by catap** [comment]
@mascguy buildbots seems idle, maybe shot this one?

**Comment by mascguy** [comment]
> @mascguy buildbots seems idle, maybe shot this one?

We might want to wait until a bit later today, until we've merged other PRs that don't take as long to build. (For example, your PR for `scala`.)

**Comment by catap** [comment]
Good point, indeed.

**Comment by mascguy** [comment]
The buildbots are still working through their backlog, as it takes hours to build `webkit-gtk` and `webkit-gtk3`. But once thy're idle, I'll merge this PR later this evening.

**Comment by catap** [comment]
Thanks, Christopher!

**Pull request merged**
